### ISSUE TITLE
'flow check -all' catches error while 'flow' or 'flow check' catches no errors

### ISSUE THREAD

**Issue by rubenhazelaar** [facebook/flow#1456]
I have some code with running flow server. `flow` gives as response `No errors!` when i add the following code to the file:

```javascript
// ... some code
/* @flow */
function add(num1, num2) {
  return num1 + num2;
}
var x = add(3, '0');
console.log(x);
// ... some code
```

Running `flow` still gives back `No errors!`. Running `flow check` gives back `Found 0 errors` 
running `flow check --all` does echo the errors (among others ofcourse). Is this normal behavior (I started experimenting with flowtype)?

**Comment by mroch** [comment]
yeah, that's expected. the `/* @flow */` applies to the whole file and has to be in a comment that's first thing in the file. so, turns out it wasn't doing anything. :/

we should add some better docs on what that comment does, and mention them in the Type Annotations guide you were following. i'm going to hijack this issue for that.

**Comment by rubenhazelaar** [comment]
Great, seems to working better now. However, when running `flow` I still get the errors from the previous run, while the echo'ed lines by flow do give back the updated line of code. Should I always run `flow check`? This is not suggested by the documentation, I believe.
